lightkube/core/client.py

```diff
@@ -176,6 +176,69 @@ def watch(self, res, *, namespace=None, labels=None, fields=None, server_timeout
         )
         return self._client.watch(br, on_error=on_error)
 
+    @overload
+    def wait(
+        self,
+        res: Type[GlobalResource],
+        name: str,
+        *,
+        for_conditions: Iterable[str] = (),
+        raise_for_conditions: Iterable[str] = (),
+    ) -> GlobalResource:
+        ...
+
+    @overload
+    def wait(
+        self,
+        res: Type[AllNamespacedResource],
+        name: str,
+        *,
+        namespace: str = None,
+        for_conditions: Iterable[str] = (),
+        raise_for_conditions: Iterable[str] = (),
+    ) -> AllNamespacedResource:
+        ...
+
+    def wait(
+        self,
+        res,
+        name: str,
+        *,
+        namespace=None,
+        for_conditions: Iterable[str] = (),
+        raise_for_conditions: Iterable[str] = (),
+    ):
+        """Waits for specified conditions."""
+
+        kind = r.api_info(res).plural
+
+        for_conditions = list(for_conditions)
+        raise_for_conditions = list(raise_for_conditions)
+
+        if not for_conditions:
+            raise ValueError("for_conditions must be specified.")
+
+        for op, obj in self.watch(res, namespace=namespace, fields={'metadata.name': name}):
+            if obj.status is None:
+                continue
+
+            if op == "DELETED":
+                raise Exception(f"{kind}/{name} was deleted while waiting for it")
```

Comment:
let's define and use a custom exception `ObjectDeleted`(?) in `core.exceptions`.